pkg/daemon/daemon.go

```diff
@@ -1290,12 +1294,67 @@ func (dn *Daemon) completeUpdate(desiredConfigName string) error {
 		return err
 	}
 
+	if err := dn.removeUpdateInProgressTaint(); err != nil {
+		return err
+	}
+
 	dn.logSystem("Update completed for config %s and node has been successfully uncordoned", desiredConfigName)
 	dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "Uncordon", fmt.Sprintf("Update completed for config %s and node has been uncordoned", desiredConfigName))
 
 	return nil
 }
 
+func (dn *Daemon) removeUpdateInProgressTaint() error {
+	// TODO: Move these 2 vars to common location to be shared with
+	var (
+		nodeUpdateBackoff = wait.Backoff{
+			Steps:    5,
+			Duration: 100 * time.Millisecond,
+			Jitter:   1.0,
+		}
+		// NodeUpdateTaint is a taint applied by MCC when the update of node starts.
+		NodeUpdateTaint = &corev1.Taint{
+			Key:    "UpdateInProgress",
+			Effect: corev1.TaintEffectPreferNoSchedule,
+		}
+	)
+	return clientretry.RetryOnConflict(nodeUpdateBackoff, func() error {
+		oldNode, err := dn.kubeClient.CoreV1().Nodes().Get(context.TODO(), dn.name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		oldData, err := json.Marshal(oldNode)
+		if err != nil {
+			return err
+		}
+
+		newNode := oldNode.DeepCopy()
+		if newNode.Spec.Taints == nil {
+			newNode.Spec.Taints = []corev1.Taint{}
+		}
+
+		taintIndex := 0
+		for i, taint := range newNode.Spec.Taints {
+			if taint.MatchTaint(NodeUpdateTaint) {
+				taintIndex = i
+			}
+		}
+
+		newNode.Spec.Taints = append(newNode.Spec.Taints[:taintIndex], newNode.Spec.Taints[taintIndex+1:]...)
```

Comment:
Alternatively, we can just iterate over taints leaving out the one we want to remove and then assign that. The complexity is similar, a bit more expensive memory wise, but easier to read, imo.